### PR TITLE
MiraMonVector fixes (2) tests improvements (+10)

### PR DESCRIPTION
**Pull request by AbelPau**
## What does this PR do?
**Fixes**

- Delete of unused function (MMInitHeader)
- Use of CPLError() in some explicit return errors

**Test improvements**

- last_version added in -lco Version
- more corrupted files tested
- unexisting coordinates or unexpected polygon construction tested datefield added in tests code simplification
- multipolygon Z added in tests
- multigeometry added in tests
- ANSI DBFEncoding test added

## Tasklist

 - [x] Make sure code is correctly formatted (cf [pre-commit configuration](https://gdal.org/development/dev_practices.html#commit-hooks))
 - [x] Add test case(s)
 - [x] All CI builds and checks have passed

### CODE REVIEW

**Comment by AbelPau** [comment]
@rouault here you have the closed one.

**Comment by rouault** [comment]
this PR lacks the added data files

**Comment by AbelPau** [comment]
> this PR lacks the added data files

yes, I'll add later. thanks

File: ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp [hunk 1]
```diff
@@ -788,6 +788,8 @@ OGRFeature *OGRMiraMonLayer::GetFeature(GIntBig nFeatureId)
                 // Get X,Y (z). MiraMon has no multipoints
                 if (MMGetGeoFeatureFromVector(phMiraMonLayer, nIElem))
                 {
+                    CPLError(CE_Failure, CPLE_AppDefined,
```
Comment (rouault) [inline ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp:791]: CPLError() messages shouldn't start or end with a newline character. Applies to below instances as well

Comment (AbelPau) [reply ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp]: Sure. Corrected. Sorry I didn't pay attention to that.
Some lost files added. It's checking now

**Comment by rouault** [comment]
backported to 3.9 per 4ad74b96820c031ace84d5ca5613df490d93166c

**Pull request merged**